scipy/interpolate/_bspl.pyx

```diff
@@ -289,7 +289,7 @@ def _colloc(const double[::1] x, const double[::1] t, int k, double[::1, :] ab,
 def _handle_lhs_derivatives(const double[::1]t, int k, double xval,
                             double[::1, :] ab,
                             int kl, int ku,
-                            const cnp.int_t[::1] deriv_ords,
+                            const cnp.npy_long[::1] deriv_ords,
```

Comment:
I think here it doesn't actually matter what is used, since presumably ords is always fine.  But the input depends on the default integer nevertheless.

I think `int` is long in cython (gah, now we are clashing with cython).  So this is good, but the site were it is created needs to ensure `long` is always used.

Reply:
> but the site were it is created needs to ensure long is always used.

I did that with `astype(int)` now, and old/new NumPy versions seems happy, but this can't possibly gel with your comment above, so I remain pretty confused at the moment.

Reply:
Yeah, I am confused.  It should break although only with NumPy main on windows.  If you want long here (32bit on win64) the only good spelling is `"long"` or `np.dtype("long")` and reusing it unfortunately.